src/library/scala/collection/immutable/VectorMap.scala

```diff
@@ -115,22 +116,26 @@ final class VectorMap[K, +V] private[immutable] (
         (this eq map) ||
           (this.size == map.size) && {
             try {
-              var i = 0
               val _size = size
-              while (i < _size) {
-                val k = fields(i)
-
-                map.get(k) match {
-                  case Some(value) =>
-                    if (!(value == underlying(k)._2)) {
-                      return false
-                    }
-                  case None =>
-                    return false
+              @tailrec
+              def loop(i: Int): Boolean = {
+                if (i < _size) {
+                  val k = fields(i)
+                  map.get(k) match {
+                    case Some(value) =>
+                      if (!(value == underlying(k)._2)) {
```

Comment:
I think this would be the same as `(value == underlying(k)._2) && loop(I+1)`, due to  short-circuiting of boolean operators.